tests/untestable/thttpclient_ssl_env_var.nim

```diff
@@ -0,0 +1,96 @@
+#
+#
+#            Nim - SSL integration tests
+#        (c) Copyright 2017 Nim contributors
+#
+#    See the file "copying.txt", included in this
+#    distribution, for details about the copyright.
+#
+## Warning: this test performs external networking.
+## Compile with:
+## ./bin/nim c -d:ssl -p:. tests/untestable/thttpclient_ssl_env_var.nim
+##
+## Test with:
+##  NIM_SSL_CERT_VALIDATION=insecure tests/untestable/thttpclient_ssl_env_var
+##  SSL_CERT_FILE=BogusInexistentFileName tests/untestable/thttpclient_ssl_env_var
+##  SSL_CERT_DIR=BogusInexistentDirName tests/untestable/thttpclient_ssl_env_var
+
+import httpclient, unittest, ospaths
+from net import newSocket, newContext, wrapSocket, connect, close, Port, CVerifyPeerUseEnvVars
+from strutils import contains
+
+const
+  expired = "https://expired.badssl.com/"
+  good = "https://google.com/"
+
+
+suite "SSL certificate check":
+
+  if existsEnv("NIM_SSL_CERT_VALIDATION"):
+    if getEnv("NIM_SSL_CERT_VALIDATION") == "insecure":
+      test "httpclient in insecure mode":
+        var ctx = newContext(verifyMode=CVerifyPeerUseEnvVars)
+        var client = newHttpClient(sslContext=ctx)
+        let a = $client.getContent(expired)
+
+      test "net socket in insecure mode":
+        var sock = newSocket()
+        var ctx = newContext(verifyMode=CVerifyPeerUseEnvVars)
+        ctx.wrapSocket(sock)
+        sock.connect("expired.badssl.com", 443.Port)
+        sock.close
+
+    else:
+      echo "ERROR: unexpected env variable value"
+      quit(1)
```

Comment:
replace all instances of `quit` with `doAssert false, optional_informative_string`
=> much friendlier for debugging when things go wrong thanks to stacktrace etc; plus doesn't break try/catch etc

Reply:
This is a test.. Maybe we don't want a stacktrace here? Because it can change with unrelated changes